### PR TITLE
 EN-1858 Modify content attribute to store and show image metadata in content (master)

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/attribute/manager/ResourceAttributeManager.java [hunk 1]
```diff
@@ -14,17 +14,57 @@
...
 /**
  * Classe manager degli attributi tipo risorsa (Image o Attach).
+ *
  * @author E.Santoboni
  */
 public class ResourceAttributeManager extends TextAttributeManager {
 
-	@Override
+    @Override
+    protected void updateAttribute(AttributeInterface attribute, AttributeTracer tracer, HttpServletRequest request) {
```
[the remark this answers is not in the document]
Comment (eugeniosant) [reply plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/attribute/manager/ResourceAttributeManager.java]: Done

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/model/extraAttribute/AbstractResourceAttribute.java [hunk 2]
```diff
@@ -134,9 +265,30 @@ public Element getJDOMElement() {
...
         }
         super.addTextElements(attributeElement);
+        this.addResourceTextElements(attributeElement, "alt", this.getResourceAltMap());
+        this.addResourceTextElements(attributeElement, "description", this.getResourceDescriptionMap());
+        this.addResourceTextElements(attributeElement, "legend", this.getResourceLegendMap());
+        this.addResourceTextElements(attributeElement, "title", this.getResourceTitleMap());
         return attributeElement;
     }
 
+    protected void addResourceTextElements(Element attributeElement, String elementName, Map<String, String> map) {
```
[the remark this answers is not in the document]
Comment (eugeniosant) [reply plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/model/extraAttribute/AbstractResourceAttribute.java]: this part is invoked whenever a text type attribute is read, even in the test phase

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/model/extraAttribute/AbstractResourceAttribute.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (joewhite101) [reply plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/model/extraAttribute/AbstractResourceAttribute.java]: Does this have specific test or assertion coverage though? Need to make sure the logic is covered to some extent.

**Pull request merged**